### PR TITLE
WIP: Beginning to update to bokeh 0.12.10 and drop previous versions.

### PR DESCRIPTION
**Pull request by ceball**
Not yet ready to review/merge.

### CODE REVIEW

**Comment by jbednar** [comment]
It all looks reasonable to me so far, thanks!

File: parambokeh/__init__.py [hunk 1]
```diff
@@ -209,7 +209,8 @@ def change_event(self):
         else:
             self._changed[p_name] = new_values
 
-        if self.p.mode == 'notebook' and self.p.push:
+        # document.hold() must have been done already? because this seems to work
```
Comment (philippjfr) [inline parambokeh/__init__.py:212]: I just checked and the bokeh ``CommsHandle`` automatically puts a hold on the document so this should be fine.

**Comment by philippjfr** [comment]
Looks good to me, would be great if we could merge this PR asap. The Action button stuff could be in a followup PR.

**Comment by philippjfr** [comment]
Thanks again for doing this @ceball, I'll merge.

**Pull request merged**